Automattic/sensei
# Show user courses for sensei_user_courses shortcode in different languages

Resolves https://github.com/Automattic/sensei/pull/7492#pullrequestreview-1978611313

## Proposed Changes
* Show user courses for sensei_user_courses shortcode in different languages.

## Known Issues

* This PR doesn't affect Course List block. Can't find a solution for it at the moment.
* Now we actually enrol a user into each translated course. The drawback of this solution is that the user is not automatically enrolled in a new translation. Could be addressed later as part of translation creation. (If it is okay, I'll create a separate issue.)

## Testing Instructions

1. On the My Courses page replace Course List block with the `[sensei_user_courses]` shortcode.
2. Translate My Courses page.
3. Follow our instructions to create one or more courses and their translations.
4. Go to the frontend of the original language and take one or more courses.
5. Switch to another language on the frontend.
6. Check that on My Courses page you see all the courses you took and their current (shared) status. (Notice: you won't see a course if it doesn't have a translation.)

## New/Updated Hooks

* `sensei_learner_enrolled_courses_query_by_progress_status_course_ids` - Filters the course IDs for a learner's enrolled courses query by progress status.
* `sensei_settings_my_course_page_id` - Filters the My Courses page ID.

## Removed Hooks (Not released yet)
- `sensei_course_manual_enrolment_enroll_learner_course_id`
- `sensei_course_manual_enrolment_withdraw_learner_course_id`

## Pre-Merge Checklist
